core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3123,9 +3123,9 @@ object ZIO extends ZIOCompanionPlatformSpecific {
 
         for {
           array <- ZIO.effectTotal(Array.ofDim[AnyRef](size))
-          queue <- Queue.bounded[(A, Int)](size)
-          _     <- queue.offerAll(as.zipWithIndex)
-          _     <- ZIO.collectAllPar_(ZIO.replicate(n)(worker(queue, array)))
+          _ <- Queue.boundedManaged[(A, Int)](size).use { queue =>
```

Comment:
My question is, without the shutdown, in the event of interruption, is everything going to be cleaned up and collected in the queue?

Reply:
Yeah so the key thing here is that shutting down a queue really just signals interruption to any fibers trying to offer or take values from the queue. It isn't as if there is some further underlying resource associated with the queue that was allocated and needs to be deallocated.

In this case the only fibers involved are the main fiber which is offering values to the queue and the fibers forked by `collectAllPar_` which are taking values from the queue and doing work. If the effect is interrupted then the main fiber is going to be interrupted and the children forked by `collectAllPar_` will be automatically interrupted, if they have been forked at all by the time interruption is observed. So I don't think there is any need to use the queue to signal interruption. I think this is a good example of why it can make sense to create a `Queue` outside the context of a `ZManaged` and it is different in some way from resources that are only valid inside the scope of a `ZManaged`.